libcalico-go/lib/validator/v3/validator.go

```diff
@@ -445,6 +447,17 @@ func validateLabels(fl validator.FieldLevel) bool {
 	return true
 }
 
+func validateSyncLabels(fl validator.FieldLevel) bool {
+	s := fl.Field().String()
+	log.Debugf("Validate SyncLabels: %s", s)
+	dst := os.Getenv("DATASTORE_TYPE")
+	if s == "Disabled" && dst == "kubernetes" {
```

Comment:
I'm not sure if it would be too extreme (@mgleung let me know if it is), but could we maybe use `strings.ToLower()` in order to still catch cases where, for example, DATASTORE_TYPE is "Kubernetes"?
(if we do go ahead with this, please remember to add test cases for it in validator_test.go)

Reply:
I just had a chat with @huiyizzz and we'll try a different approach. I don't think that we can guarantee that the env variable will be set each time for every component that calls this so we'll try something that doesn't depend upon this value.

Reply:
`strings.ToLower()` would be great to use here though normally to ensure we're matching properly though.